src/ln/channelmonitor.rs

```diff
@@ -609,14 +782,59 @@ impl ChannelMonitor {
 	}
 
 	pub(super) fn unset_funding_info(&mut self) {
-		self.funding_txo = None;
+		self.key_storage = match self.key_storage {
+			Storage::Local { ref revocation_base_key, ref htlc_base_key, ref delayed_payment_base_key, ref payment_base_key, ref shutdown_pubkey, ref prev_latest_per_commitment_point, ref latest_per_commitment_point, short_channel_id, .. } => {
+				Storage::Local {
+					revocation_base_key: *revocation_base_key,
+					htlc_base_key: *htlc_base_key,
+					delayed_payment_base_key: *delayed_payment_base_key,
+					payment_base_key: *payment_base_key,
+					shutdown_pubkey: *shutdown_pubkey,
+					prev_latest_per_commitment_point: *prev_latest_per_commitment_point,
+					latest_per_commitment_point: *latest_per_commitment_point,
+					funding_info: None,
+					short_channel_id: short_channel_id,
+				}
+			},
+			Storage::Watchtower { .. } => {
+				unimplemented!();
+			},
+		}
+	}
+
+	pub(super) fn unset_short_channel_id(&mut self) {
```

Comment:
This is never called?

Reply:
Removed, as not justification contrary to unset_funding_txo